### PR TITLE
Update type conversion from pandas to timestamp to support various the timestamp types

### PR DESCRIPTION
**Pull request by achals**
**What this PR does / why we need it**:
Currently, Inferring features fails on columns of type timestamp[ms, tz=UTC] or other tz. This diff attempts to fix using a regex instead of enumerating all the different possibilities. 

An option is to use the pandas type instead of the string representation, but this is probably good enough.
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timestamp.html

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Fix Inferring features columns of pandas timestamp types.
```

### CODE REVIEW

File: sdk/python/feast/type_map.py [hunk 1]
```diff
@@ -438,8 +439,14 @@ def pa_to_value_type(pa_type: object):
 
 
 def pa_to_feast_value_type(value: Union[pa.lib.ChunkedArray, str]) -> ValueType:
+    value_type = (
+        value.type.__str__() if isinstance(value, pa.lib.ChunkedArray) else value
```
Comment (woop) [inline sdk/python/feast/type_map.py:443]: Would you mind explaining this line?

Comment (achals) [reply sdk/python/feast/type_map.py:443]: This is moved from line 459 - basically this converts the type of a pandas column into a string representation, if the stringified version has not already been passed in.

**Review by woop — APPROVED** [review]

**Pull request merged**